examples/ninepatch.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+"""pygame.examples.ninepatch
+Demonstrate the purpose of the 9-patch scale method and a way to implement it.
+
+9-patch scaling allows us to scale a surface while keeping 4 corners intact.
+This is very useful for GUI applications.
+
+How it works:
+The 4 corners are extracted from the original surface and pasted on the new surface.
+The 4 sides in between the corners are scaled horizontally or vertically to fit
+between the new surface corners.
+The central part of the image is scaled and pasted.
+
+In this example a green fill is used which can be done faster and easier with pygame.draw.rect
+but usually you'd have a custom surface that needs intact corners.
+"""
+
+import os
+import pygame
+import typing
+
+SCREEN_SIZE = pygame.Vector2(600, 500)
+SCALE_SIZE = pygame.Vector2(500, 150)
+INNER_SCALE_SIZE = pygame.Vector2(400, 115)
+CORNER_SIZE = 20
+INNER_CORNER_SIZE = 7
+
+
+def ninepatch_scale(
+    surface: pygame.Surface,
+    size: typing.Sequence[int],
+    corner_size: int,
+    alpha: bool = True,
+    smooth: bool = False,
+    dest_surface: pygame.Surface | None = None,
+) -> pygame.Surface:
+    # the code of this function also emulates how a transform-like ninepatch function would likely behave
+    if dest_surface is not None:
+        ret_surface = dest_surface
+        if ret_surface.size != size:
+            raise ValueError("Destination surface doesn't match the provided size")
+    else:
+        ret_surface = pygame.Surface(
+            size, pygame.SRCALPHA * alpha
+        )  # when alpha is False the flags become 0
+
+    # aliases
+    ret_w, ret_h = size  # non-sequence argument catched by python
+    src_w, src_h = surface.size
+    c = corner_size
+    scale_func = pygame.transform.smoothscale if smooth else pygame.transform.scale
+    subsurface = surface.subsurface  # alias for performance/clarity
+
+    if corner_size < 0 or corner_size > min(ret_w, ret_h):
+        raise ValueError(
+            "Corner size must be nonnegative and not greater than the smaller between width and height"
+        )
+    if corner_size == 0:  # default to normal scaling
+        return scale_func(surface, size)
+
+    src_x_side, ret_x_side = src_w - c * 2, ret_w - c * 2
+    src_y_side, ret_y_side = src_h - c * 2, ret_h - c * 2
```

Comment:
```suggestion
    dest_size = size  # non-sequence argument catched by python
    src_size = surface.size
    c = corner_size
    scale_func = pygame.transform.smoothscale if smooth else pygame.transform.scale
    subsurface = surface.subsurface  # alias for performance/clarity

    if not 0 < corner_size < min(dest_size):
        raise ValueError(
            "Corner size must be positive and less than the smallest between destination width and destination height"
        )
    if corner_size == 0:  # default to normal scaling
        return scale_func(surface, size)
        
    ret_x_side, ret_y_side = src_size - pygame.Vector2(c) * 2
    src_x_side, src_y_side = ret_size - pygame.Vector2(c) * 2    
```
For these 2 lines : 
```py
    ret_x_side, ret_y_side = src_size - pygame.Vector2(c) * 2
    src_x_side, src_y_side = ret_size - pygame.Vector2(c) * 2  
```
It might be not beautiful, but this is because you are not currently supporting custom x and y corners. I would like to recommend this additional feature.

For beginners mental health, you should comment what the variable you created are used for.

Reply:
bro wants me to suffer even more 😭 I'm about to take back the idea of adding this in the examples 😭

Reply:
After a small discussion on discord, I agree it's not necessary.

Reply:
I will improve what you said to improve in this review comment tho